src/js/background/context-menu.js

```diff
@@ -0,0 +1,388 @@
+        }
+      );
+    }
+  },
+  menus: {
+    create: async (data, opts=null) => {     
+      // If multiple items need to be created: 
+      if (Array.isArray(data)) {
+        data.forEach(async (menu) => {
+          await browser.menus.create(menu, relayContextMenus.utils.onCreatedCallback);
+        });
+
+        return;
+      }
+
+      // Loop Through Existing Aliases
+      if (opts?.createExistingAliases) {
+        
+        // TODO: Edgecase Fix if API doesn't respond (use local)
+        const aliases = await relayContextMenus.utils.getAliases();
+        
+        const filteredAliases = opts.currentWebsite
+          ? relayContextMenus.utils.getSiteSpecificAliases(
+              aliases,
+              opts.currentWebsite
+            )
+          : relayContextMenus.utils.getMostRecentAliases(aliases);
+
+        // // Cache server set locally
+        // browser.storage.local.set({ relayAddresses: aliases });
+        const currentUseAliasIds = [];
+
+        for (const alias of filteredAliases) {
+          const title = alias.description ? alias.description : alias.address;
+          const id = "fx-private-relay-use-existing-alias_" + alias.id;
+          
+          currentUseAliasIds.push(id);
+          
+          data.title = title;
+          data.id = id;
+          data.parentId = opts.parentId;
+          await browser.menus.create(data, relayContextMenus.utils.onCreatedCallback);
+        }
+        
+        await browser.menus.refresh();
+
+        // Used to parse "use alias" context menus
+        await browser.storage.local.set({ currentUseAliasIds });
+
+        return;
+      }
+
+      await browser.menus.create(data, relayContextMenus.utils.onCreatedCallback);
+      
+    },
+    remove: async (id) => {
+      await browser.menus.remove(id);
+    }
+  }, 
+  utils: {
+    getAliases: async () => {
+
+      // TODO: This requires network access/server uptime. 
+      // Is there a way to test/fallback to local?
+      if ( await getServerStoragePref() ) {
+        return await getAliasesFromServer();
+      }
+
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      return relayAddresses;
+
+    },
+    getGeneratedForHistory: async (website) => {
+      const { relayAddresses } = await browser.storage.local.get("relayAddresses");
+      
+      function checkAvailability(aliases, domain) {
+        return aliases.some(function(alias) {
+            return domain === alias.generated_for;
+        });
+      }
+
+      const websiteMatch = checkAvailability(relayAddresses, website)
+
+      return websiteMatch
+    },
+    getHostnameFromRegex: (url) => {
+      // TODO: Unused function
+      // https://stackoverflow.com/a/54947757
+      // run against regex
+      const matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
+      // extract hostname (will be null if no match is found)
+      return matches && matches[1];
+    },  
```

Comment:
If it's unused, might as well remove it? Feels more brittle than deferring to the browser for this anyway.

```suggestion
```

Reply:
Removed in 787f56125a7bdf2aa0bd51eb60f64daecf080440